app/javascript/onboarding/components/IntroSlide.jsx

```diff
@@ -2,27 +2,98 @@ import { h, Component } from 'preact';
 import PropTypes from 'prop-types';
 
 import Navigation from './Navigation';
-import { userData, updateOnboarding } from '../utilities';
+import { getContentOfToken, userData, updateOnboarding } from '../utilities';
 
+/* eslint-disable camelcase */
 class IntroSlide extends Component {
   constructor(props) {
     super(props);
 
+    this.handleChange = this.handleChange.bind(this);
     this.onSubmit = this.onSubmit.bind(this);
     this.user = userData();
+
+    this.state = {
+      checked_code_of_conduct: false,
+      checked_terms_and_conditions: false,
+      text: null,
+    };
   }
 
   componentDidMount() {
-    updateOnboarding('intro slide');
+    updateOnboarding('v2: intro, code of conduct, terms & conditions');
   }
 
   onSubmit() {
     const { next } = this.props;
-    next();
+    const csrfToken = getContentOfToken('csrf-token');
+
+    fetch('/onboarding_checkbox_update', {
+      method: 'PATCH',
+      headers: {
+        'X-CSRF-Token': csrfToken,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ user: this.state }),
+      credentials: 'same-origin',
+    }).then((response) => {
+      if (response.ok) {
+        localStorage.setItem('shouldRedirectToOnboarding', false);
+        next();
+      }
+    });
+  }
+
+  handleChange(event) {
+    const { name } = event.target;
+    this.setState((currentState) => ({
+      [name]: !currentState[name],
+    }));
+  }
+
+  handleShowText(event, id) {
+    event.preventDefault();
+    this.setState({ text: document.getElementById(id).innerHTML });
+  }
+
+  buttonIsDisabled() {
```

Comment:
Consider renaming this.
```suggestion
  isButtonDisabled() {
```